Python/Product/TestAdapter.Executor/Pytest/TestExecutorPytest.cs

```diff
@@ -132,22 +129,46 @@ IFrameworkHandle frameworkHandle
             IFrameworkHandle frameworkHandle
         ) {
             PythonProjectSettings settings = testGroup.Key;
-            if (settings.TestFramwork != TestFrameworkType.Pytest) {
+            if (settings == null || settings.TestFramwork != TestFrameworkType.Pytest) {
                 return;
             }
 
             using (var executor = new ExecutorService(settings, frameworkHandle, runContext)) {
+                Dictionary<string, TestResult> idToResultsMap = CreatePytestIdToVsTestResultsMap(testGroup);
                 var resultsXML = executor.Run(testGroup);
 
-                var testResults = TestResultParser.Parse(resultsXML, testGroup);
-                foreach (var result in testResults) {
+                //Read pytest results from xml
+                if (File.Exists(resultsXML)) {
```

Comment:
This will silently do nothing if the test result xml file wasn't created. Log an error?

Reply:
you're right..

Reply:
added an error log